botorch/optim/optimize.py

```diff
@@ -1239,11 +1306,54 @@ def optimize_acqf_discrete_local_search(
     inequality_constraints = inequality_constraints or []
     for i in range(q):
         # generate some starting points
-        if i == 0 and batch_initial_conditions is not None:
-            X0 = _filter_invalid(X=batch_initial_conditions.squeeze(1), X_avoid=X_avoid)
-            X0 = _filter_infeasible(
-                X=X0, inequality_constraints=inequality_constraints
-            ).unsqueeze(1)
+        if i == 0:
+
+            if batch_initial_conditions is not None:
```

Comment:
same here - a lot of stuff happening here, can you factor this out into a helper function so this is easier to read?

Reply:
Actually I think that I have the wrong logic here as the size of the initial_batch_constraints is compared to `n_repeats` not `raw_samples`.